saleor/discount/tests/test_discounts.py

```diff
@@ -557,3 +560,109 @@ def test_fetch_catalogue_info_for_sale_has_one_element_sets(sale):
     assert catalogue_info["collections"] == collection_ids
     assert catalogue_info["products"] == product_ids
     assert catalogue_info["variants"] == variant_ids
+
+
+def test_get_discount_name_only_rule_name(promotion):
+    # given
+    promotion.name = ""
+    promotion.save(update_fields=["name"])
+
+    rule = promotion.rules.first()
+
+    # when
+    name = get_discount_name(rule, promotion)
+
+    # then
+    assert name == rule.name
+
+
+def test_get_discount_name_only_rule_promotion_name(promotion):
+    # given
+    rule = promotion.rules.first()
+    rule.name = ""
+    rule.save(update_fields=["name"])
+
+    # when
+    name = get_discount_name(rule, promotion)
+
+    # then
+    assert name == promotion.name
+
+
+def test_get_discount_name_rule_and_promotion_name(promotion):
+    # given
+    rule = promotion.rules.first()
+
+    # when
+    name = get_discount_name(rule, promotion)
+
+    # then
+    assert name == f"{promotion.name}: {rule.name}"
+
+
+def test_get_discount_name_empty_names(promotion):
+    # given
+    rule = promotion.rules.first()
+
+    rule.name = ""
+    rule.save(update_fields=["name"])
+
+    promotion.name = ""
+    promotion.save(update_fields=["name"])
+
+    # when
+    name = get_discount_name(rule, promotion)
+
+    # then
+    assert name == ""
+
+
+def test_get_discount_translated_name_only_rule_translation(rule_info):
+    # given
+    rule_info_data = rule_info._asdict()
+    rule_info_data["promotion_translation"] = None
```

Comment:
Hmm, shouldn't
```
rule_info.promotion_translation = None
```

do the same thing? I might not understand how `promotion_translation` value is filled and trying to grasp that.

Reply:
`rule_info` is `NamedTuple` so it's immutable and I cannot do it in simpler way :(